supernode.cpp

```diff
@@ -1,28 +1,60 @@
 #include <stdexcept>
 #include <cmath>
 #include <boost/multiprecision/cpp_int.hpp>
-#include <omp.h>
 #include "include/supernode.h"
 #include "include/util.h"
 #include "include/graph_worker.h"
 
-Supernode::Supernode(uint64_t n, long seed): sketches(log2(n)), idx(0), logn(log2
-(n)), seed(seed), n(n) {
+uint32_t Supernode::bytes_size;
+
+Supernode::Supernode(uint64_t n, long seed): idx(0), logn(log2(n)),
+					     n(n), seed(seed), sketch_size(Sketch::sketchSizeof()) {
+
   // generate logn sketches for each supernode (read: node)
-  for (int i=0;i<logn;++i)
-    sketches[i] = new Sketch(n*n, seed++);
+  for (int i = 0; i < logn; ++i) {
+    Sketch::makeSketch(get_sketch(i), seed++);
+  }
+}
+
+Supernode::Supernode(uint64_t n, long seed, std::fstream &binary_in) :
+  idx(0), logn(log2(n)), n(n), seed(seed), sketch_size(Sketch::sketchSizeof()) {
+
+  // read logn sketches from file for each supernode (read: node)
+  for (int i = 0; i < logn; ++i) {
+    Sketch::makeSketch(get_sketch(i), seed++, binary_in);
+  }
+}
+
+Supernode::SupernodeUniquePtr Supernode::makeSupernode(uint64_t n, long seed) {
+  Sketch::n = n * n;
+  set_size(n);
```

Comment:
this should not happen in this constructor. this will be called `n - 1` times more than it should.

Reply:
I don't know if this will make much of a difference because as previously discussed we only use this call at the beginning of the program. So yes our initialization might take a tiny bit longer but our ingestion rate will be unaffected.

There is benefit to making the process of setting the size of a super node transparent to the caller.

Reply:
I would disagree. We should only have one (fixed) value of `n` and fixed values of `Sketch::n` and sketch size. Will it require more architectural changes to kick this out of supernode constructor?

Reply:
I guess you're right. I think what we would want to do is have a static configure function for Sketch, Supernode, and Graph which sets these parameters appropriately. 

We can also set it up so calling the Graph configure function calls super node which calls Sketch

I'll code that up and push it.